### PR TITLE
Improve spec language around identity keys, resolution, gateways, and TLD

### LINKED ISSUE #130: Add guidance for recommending fully qualified URIs for all properties, not just keys
the guidance is currently only present around keys, but I'd like to update this to recommend fully qualified identifiers for all properties https://did-dht.com/#representing-keys, I'll open an issue for this.

generally I think we should avoid fragments with # since - at least to me - they only make sense in the context of a full identifier

_Originally posted by @decentralgabe in https://github.com/TBD54566975/web5-js/issues/413#issuecomment-1947501047_

### PR DESCRIPTION
**Pull request by thereisnogabe**
A number of clarifying adjustments.

Also fix https://github.com/TBD54566975/did-dht-method/issues/130

### CODE REVIEW

File: spec/spec.md [hunk 1]
```diff
@@ -423,21 +448,22 @@ To create a `did:dht`, the process is as follows:
...
 
-1. Take the suffix of the DID, that is, the _[[ref:z-base-32]] encoded identifier key_, and pass it to a [[ref:Pkarr]]
+1. Take the [[ref:suffix]] of the DID, that is, the _[[ref:z-base-32]] encoded identifier key_, and submit it to a [[ref:Pkarr]]
 relay or a [[ref:Gateway]].
 
 2. Decode the resulting [[ref:BEP44]] response's `v` value using [[ref:bencode]].
 
 3. Reverse the DNS [property mapping](#property-mapping) process and re-construct a conformant [[ref:DID Document]].
 
+    a. Expand all identifiers to their fully-qualified form (e.g. `did:dht:uodqi99wuzxsz6yx445zxkp8ddwj9q54ocbcg8yifsqru45x63kj#0`
```
Comment (thereisnogabe) [inline spec/spec.md]: @frankhinek addressing https://github.com/TBD54566975/did-dht-method/issues/130 here

**Review by csuwildcat — APPROVED** [review]

File: spec/spec.md [hunk 2]
```diff
@@ -79,50 +79,52 @@ The keywords MAY, MUST, MUST NOT, RECOMMENDED, SHOULD, and SHOULD NOT in this do
...
 [[def:DID Suffix, Suffix]]
-~ The unique identifier string within a DID URI. e.g. The unique suffix of `did:dht:123` would be `123`.
-
-[[def:DID Suffix Data, Suffix Data]]
-~ Data required to deterministically generate a DID, the [[ref:Identity Key]].
+~ The unique identifier string within a DID URI. For DID DHT the suffix is the [[ref:z-base-32]] encoded [[ref:Identity Key]].
 
 [[def:Identity Key]]
-~ An [[ref:Ed25519]] public key encoded with [[ref:z-base-32]] used to uniquely identify a `did:dht` document.
+~ An [Identity Key](#identity-key) is a [[ref:Ed25519]] public key encoded required to authenticate all records in
```
Comment (mihai-chiorean) [inline spec/spec.md]: this reads weird. `is a [[ref:Ed25519]] public key encoded required` specifically. `are you missing `with [[ref:z-base-32]]`?

File: spec/spec.md [hunk 3]
```diff
@@ -151,25 +153,54 @@ Alternatively, one can interpret the encoding rules as a series of transformatio
...
-| --------- | ---- | ------ | ---------------------------------------- |
-| _did.     | TXT  |  7200  | v=0;vm=k0,k1,k2;auth=k0;asm=k1;inv=k2;del=k2;srv=s0,s1,s2 |
-| _k0._did. | TXT  |  7200  | id=0;t=0;k=`<unpadded-b64url>`           |
-| _k1._did. | TXT  |  7200  | id=1;t=1;k=`<unpadded-b64url>`           |
-| _k2._did. | TXT  |  7200  | id=2;t=1;k=`<unpadded-b64url>`           |
-| _s0._did. | TXT  |  7200  | id=domain;t=LinkedDomains;se=foo.com     |
-| _s1._did. | TXT  |  7200  | id=dwn;t=DecentralizedWebNode;se=https://dwn.tbddev.org/dwn5 |
+| Name         | Type | TTL    | Rdata                                                        |
+| ------------ | ---- | ------ | ------------------------------------------------------------ |
+| _did.`<ID>`. | TXT  |  7200  | v=0;vm=k0,k1,k2;auth=k0;asm=k1;inv=k2;del=k2;srv=s0,s1,s2    |
```
Comment (mihai-chiorean) [inline spec/spec.md:189]: yess. I was confused by this initially. seems obvious in hindsight, but it's good to have it be explicit

File: spec/spec.md [hunk 4]
```diff
@@ -383,20 +409,19 @@ A sample transformation of a fully-featured DID Document to a DNS packet is exem
...
+| _k0._did.    | TXT  | 7200  | t=0;k=afdea69c63605863a68edea0ff7ff49dde0a96ce7e9249eb7780dd3d6f2ab5fc             |
+| _k1._did.    | TXT  | 7200  | t=1;k=AyiNAz7y-XBr853PBAzgAOU_c0Hyw0Gb69Hr9jTC3MQ8                                 |
+| _s0._did.    | TXT  | 7200  | id=dwn;t=DecentralizedWebNode;se=https://example.com/dwn1,https://example.com/dwn2 |
 
 ### Operations
 
-Entries to the [[ref:DHT]] require a signed record. As such, the keypair used for the [[ref:Pkarr]]
-identifier is also used to sign the [[ref:DHT]] record. This keypair ****MUST**** always be present in a `did:dht` document
-and is referred to as the [[ref:Identity Key]].
+Entries to the [[ref:DHT]] require a signed record as per [[ref:BEP44]]. As such, the keypair used for the [[ref:Pkarr]]
```
Comment (mihai-chiorean) [inline spec/spec.md:423]: really good to see this called out

File: spec/spec.md (hunk 1 above)
Comment (mihai-chiorean) [reply spec/spec.md]: when this calls out "all identifiers", just so I understand, isn't only the Identity Key represented as "#0" or "0"? also this refers to verification methods, right? are there other parts of the did doc that could be represented as `#xxx`?

File: spec/spec.md (hunk 2 above)
Comment (thereisnogabe) [reply spec/spec.md]: yes good call out - thanks!

File: spec/spec.md (hunk 1 above)
Comment (thereisnogabe) [reply spec/spec.md]: I mean to say a DID Doc should always use fully qualified identifiers, regardless of the verification method id

it is confusing to handle `#0` or `#key-1` as opposed to `did:dht:example#0` or `did:dht:example#key-1`

makes it tougher for implementations to handle some cases with fragments and others without

**Pull request merged**